src/element/textElement.ts

```diff
@@ -288,62 +288,36 @@ export const measureText = (
   return { width, height };
 };
 
-// Using this factor to align with most of the browsers and legacy algorithm
-const LINE_HEIGHT_FACTOR = 1.25;
-
-export const computeNextLineHeightForText = (
-  originalElement: NonDeleted<ExcalidrawTextElement>,
-  updatedElement: NonDeleted<ExcalidrawTextElement>,
-) => {
-  const originalLineHeight = originalElement.lineHeight;
-
-  // Calculate line height relative to font size
-  if (
-    Math.abs(
-      originalLineHeight - originalElement.fontSize * LINE_HEIGHT_FACTOR,
-    ) < Number.EPSILON
-  ) {
-    return updatedElement.fontSize * LINE_HEIGHT_FACTOR;
-  }
-  return getLegacyLineHeightForText(updatedElement);
-};
-
 /**
  * To maintain backward compatibility with old diagrams where the line height
  * was browser-dependent, calculating the line height based on the height of the
  * text element and the number of lines gives us a pretty accurate
  * original line height.
  */
-export const getLegacyLineHeightForText = (
-  textElement: ExcalidrawTextElement,
-) => {
+export const detectLineHeight = (textElement: ExcalidrawTextElement) => {
   const lineCount = splitInToLines(textElement.text).length;
-  return textElement.height / lineCount;
+  return (textElement.height /
+    lineCount /
+    textElement.fontSize) as ExcalidrawTextElement["lineHeight"];
 };
 
 /**
- * We calculate the line height from the font size and the line height factor,
+ * We calculate the line height from the font size and the unitless line height,
  * aligning with the W3C spec.
  */
-export const getApproxLineHeight = (fontSize: number) => {
-  // Calculate line height relative to font size
-  return fontSize * LINE_HEIGHT_FACTOR;
-};
-
-export const getApproxMinLineHeight = (fontSize: number) => {
-  return getApproxLineHeight(fontSize) + BOUND_TEXT_PADDING * 2;
+export const getLineHeightInPx = (
+  fontSize: ExcalidrawTextElement["fontSize"],
+  lineHeight: ExcalidrawTextElement["lineHeight"],
+) => {
+  return fontSize * lineHeight;
 };
 
-export const computeMinHeightForBoundText = (
-  originalElement: NonDeleted<ExcalidrawTextElement>,
-  updatedElement: NonDeleted<ExcalidrawTextElement>,
+// FIXME rename to getApproxMinContainerHeight
+export const getApproxMinLineHeight = (
```

Comment:
kept the name to reduce conflicts with https://github.com/excalidraw/excalidraw/pull/6287 as mentioned, but though PR will still have conflicts regardless